src/awx_plugins/credentials/aws_assumerole.py

```diff
@@ -0,0 +1,170 @@
+
+import datetime
+import hashlib
+
+import boto3
+from botocore.exceptions import ClientError
+
+from .plugin import CredentialPlugin, translate_function as _
+
+
+_aws_cred_cache = {}
+
+
+assume_role_inputs = {
+    'fields': [
+        {
+            'id': 'access_key',
+            'label': _('AWS Access Key'),
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS access key for'
+                'the user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'secret_key',
+            'label': 'AWS Secret Key',
+            'type': 'string',
+            'secret': True,
+            'help_text': _(
+                'The optional AWS secret key for the'
+                'user who will assume the role.',
+            ),
+        },
+        {
+            'id': 'external_id',
+            'label': 'External ID',
+            'type': 'string',
+            'help_text': _(
+                'The optional External ID which will'
+                'be provided to the assume role API.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'identifier',
+            'label': 'Identifier',
+            'type': 'string',
+            'help_text': _(
+                'The name of the key in the assumed AWS role'
+                'to fetch [AccessKeyId | SecretAccessKey | SessionToken].',
+            ),
+        },
+    ],
+    'required': [
+        'role_arn',
+    ],
+}
+
+
+def aws_assumerole_getcreds(
+        access_key: str | None,
+        secret_key: str | None,
+        role_arn: str | None,
+        external_id: int,
+) -> dict:
+    """Return the credentials for use.
+
+    :param access_key: The AWS access key ID.
+    :type access_key: str
+    :param secret_key: The AWS secret access key.
+    :type secret_key: str
+    :param role_arn: The ARN received from AWS.
+    :type role_arn: str
+    :param external_id: The external ID received from AWS.
+    :type external_id: int
+    :returns: The credentials received from AWS.
+    :rtype: dict
+    :raises ValueError: If the client response is bad.
+    """
+    from mypy_boto3_sts.client import STSClient
+
+    connection: STSClient = boto3.client(
+        service_name='sts',
+        # The following EE creds are read from the env if they are not passed:
+        aws_access_key_id=access_key,  # defaults to `None` in the lib
+        aws_secret_access_key=secret_key,  # defaults to `None` in the lib
+    )
+    try:
+        from mypy_boto3_sts.type_defs import AssumeRoleResponseTypeDef
+
+        response: AssumeRoleResponseTypeDef = connection.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName='AAP_AWS_Role_Session1',
+            ExternalId=external_id,
+        )
+    except ClientError as client_err:
+        raise ValueError(
+            f'Got a bad client response from AWS: {ce.message}.',
```

Comment:
The linter caught this: https://github.com/ansible/awx-plugins/actions/runs/11226550925/job/31207297582?pr=24#step:18:83
```suggestion
            f'Got a bad client response from AWS: {client_err.message}.',
```